### PR TITLE
Standalone generate docs

### PR DESCRIPTION
**Pull request by stefanor**
## Summary of changes

Allow `scripts/generate_docs.py` to be run directly on the source tree (with `PYTHONPATH=src`, for example).
This makes it easier to run during a Debian package build.

While we're here, update the makefile, the `docs` target wasn't updated after a noxfile reorg.

## Test plan
Tested by running
```
$ PYTHONPATH=src python3 scripts/generate_docs.py
$ make docs
```

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -13,7 +13,7 @@ publish:
 	pipx run nox -s publish
 
 docs:
-	pipx run nox -s docs -r
+	pipx run nox -s build_docs -r
```
Comment (uranusjr) [inline makefile:16]: Hmm does this indicate nobody is actually using the makefile…?

**Review by uranusjr — APPROVED** [review]

File: makefile (hunk 1 above)
Comment (cs01) [reply makefile:16]: I think everything in it was migrated to nox. If it wasn’t, it should be, and the makefile should be deleted.

**Review by gaborbernat — CHANGES_REQUESTED** [review]
The CI is failing here.

**Comment by cs01** [comment]
The makefile is now deprecated. The noxfile is the only way to guarantee (or at least come close to guaranteeing) a reproducible build with the same dependencies and environment (the pipx binary is on the user's path because `DOC_DEPENDENCIES = [".", "jinja2", "mkdocs", "mkdocs-material"]` contains `.` which installs pipx into the virtual environment nox runs the task in).

**Pull request closed without merging**